### PR TITLE
[next-pwa] add forward decl for Location

### PR DESCRIPTION
**Pull request by rbuckton**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] ~~Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>~~
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~

This fixes a build break in the DT nightly pipeline due to a missing declaration for `Location` that a dependency expects. This also changes the forward declarations for `PopStateEventInit` and `PopStateEvent` to match the ones in `lib.dom.generated.d.ts`.

### CODE REVIEW

File: types/next-pwa/package.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
   "private": true,
   "dependencies": {
+    "@types/node": "*",
+    "@types/react": "*",
+    "@types/react-dom": "*",
```
Comment (jakebailey) [inline types/next-pwa/package.json:6]: If you copied these from my DT monorepo branch, I don't think they are needed here; in `master` we're still using `typesRoot` so these won't have any effect.

File: types/react/package.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
   "private": true,
   "dependencies": {
-    "csstype": "^3.0.2"
+    "csstype": "^3.0.2",
+    "definitely-typed": "file:../.."
```
Comment (jakebailey) [inline types/react/package.json:5]:
```suggestion
    "csstype": "^3.0.2"
```

File: types/next-pwa/package.json (hunk 1 above)
Comment (rbuckton) [reply types/next-pwa/package.json:6]: Yeah, there seems to be an issue with `typesRoot` resolution. I'm looking into it.

**Review by jakebailey — APPROVED** [review]

**Comment by NivaldoFarias** [comment]
only pointing this out in case it conflicts, but #66899 also fixes this with a different approach

**Comment by rbuckton** [comment]
I'm wary of pulling in DOM types here. The index.d.ts explicitly calls out incompatibility with DOM and webworker libs, and the types were initially provided pointing to webworker.

**Review by sandersn — APPROVED** [review]

**Comment by rbuckton** [comment]
> only pointing this out in case it causes conflicts, but #66899 also fixes this (with a different approach)

Was there a reason to use `"webworker"` instead of `"dom"` when this was originally added?

**Comment by NivaldoFarias** [comment]
> Was there a reason to use `"webworker"` instead of `"dom"` when this was originally added?

I dug up a bit and found this [comment](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63391#issuecomment-1328129756) I made a year ago that mentions this issue, which would break the declarations at that time (ts 5.1). 

However by the looks of it, if all package specs are tested and they work fine, there shouldn't be a problem in favouring PR #66899.

**Comment by rbuckton** [comment]
Closing in favor of #66899

**Pull request closed without merging**